### ISSUE TITLE
question regarding to the settings and evaluating the training

### ISSUE THREAD

**Issue by waynezw0618** [cselab/smarties#4]
Hi Novatig:
thanks for replying. the updated readme is helpful. Simultaneously I have read your paper of glider, seem the control result is sensitive to the training method and its setting. now I implement the a boat navigation case based on the boatNav.  everything works fine with runTest for me. then I use RACE to train, the case is to control  the boat to turning some angle and move to another location without limitation the final nose angle and termination speed in the reward. but I get quite little success sample during training. most of the time the boat got lost, does that influent the training? would you please make some suggestion? RACE? DQN? PPO? or change some setting?
furthermore,  after training, I want to evaluate solution by restart, I use the command  "boatNav --restart /public3/home/sc51719/soft/smarties/apps/boadNav/evalue RACE.json " but I get many trajectories. which i suppose to be one, since I suppose restart is going to load the trained network to evaluate the performance of one task.  
can you provide me some suggestion?

**Comment by novatig** [comment]
Hi,

I am not familiar with the boatNav environment, it was not implemented by me.
If you want I could ask the developer to update the source files. Otherwise, I should just take them out of the repo.

For the rest, I don't quite get your questions. Algorithms-wise I typically use VRACER as default, just because it combines experience replay and policy gradients. The readme contains some default values for the hyper-parameters which depend on problem size.

The option you should use how many evaluation sims to run is ``smarties.py [...] --nEvalEpisodes 1``.

If the behavior is not as expected: does smarties say that restart was successful? What do you mean by lost?

**Comment by waynezw0618** [comment]
> The option you should use how many evaluation sims to run is `smarties.py [...] --nEvalEpisodes 1`.
so it is not the option '--restart ' to run the evaluation?  or I can run "boatNav --restart /public3/home/sc51719/soft/smarties/apps/boadNav/evalue RACE.json --nEvalEpisodes 1" the same thing?
for boatNav case, the target is to move the boat from the start to the end by adjusting the torque and thrust, there are two situations, either arrived or not.  for my case, during the evaluation I get lots of results but non of them arrives at the terminal in the evaluation .

**Comment by novatig** [comment]
Sorry for the wait.

Again, I am afraid you should look into the boatNav case yourself because the person who developed it has left the lab.
On a recent commit I replaced the default hyperparams I used for the refer paper with hyperparams that change with problem size, that might be a better initial guess.

As far as running with ./boatNav instead with the smarties.py script.
Yes the settings shoul be the same. You can check with ./boatNav --help
Hoever you cannot specify the settings .json file from the executable. smarties will directly use the file it finds in the run directory.
Try removing RACE.json (why not the provided RACER.json? or the default VRACER.json?) and see if the args are parsed correctly.

**Comment by waynezw0618** [comment]
I rewrite the environment part and tested. I actually don't have any setting.json like VRACER.json in my run folder. seems it reads from the somewhere I don't know. whether we can open a private  repo to discuss this ?
